### PR TITLE
People button with role permissions

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

Add role permissions to People button

# Why

People button needs to be disabled for the 'Consumer' role when room calls is released

# How Tested

Local sample testing.
image

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/components/buttons/People.tsx [hunk 1]
```diff
@@ -15,6 +17,14 @@ const icon = (): JSX.Element => <CallCompositeIcon iconName={'ControlButtonParti
 /** @beta */
 export const People = (props: ControlBarButtonProps): JSX.Element => {
   const { strings, onRenderOnIcon, onRenderOffIcon, onClick } = props;
+
+  /* @conditional-compile-remove(rooms) */
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/components/buttons/People.tsx:21]: This line shouldn't be conditionally complied?

Comment (mgamis-msft) [reply packages/react-composites/src/composites/CallComposite/components/buttons/People.tsx:21]: We are spreading props directly to ControlButton (on line 45) in stable. So isn't this line is not needed in stable?

Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/components/buttons/People.tsx:21]: I think we only need to conditionally compile the permissions statement `disabled = disabled || !_usePermissions().participantList`

Comment (mgamis-msft) [reply packages/react-composites/src/composites/CallComposite/components/buttons/People.tsx:21]: But the line: 
```let disabled = props.disabled;```
is not needed in stable.

Comment (PorterNan) [reply packages/react-composites/src/composites/CallComposite/components/buttons/People.tsx:21]: Got it, so it's all beta feature even disable prop

**Review by PorterNan — APPROVED** [review]

**Review by prabhjot-msft — APPROVED** [review]

**Pull request merged**